typescript/src/schema/api.ts

```diff
@@ -0,0 +1,41 @@
+import * as plugin from '../../../pbjs-genfiles/plugin';
+
+import {Naming} from './naming';
+import {Proto} from './proto';
+
+export type ProtosMap = {
+  [filename: string]: Proto
+};
+
+export class API {
+  naming: Naming;
+  protos: ProtosMap;
+  // TODO: subpackages
```

Comment:
what's that mean?

Reply:
I think for some API surfaces, it has some subpackages for better "organizing" the files. For building process, we have to include those files. So we will have subpackage_view as a property later as python_generator does.

Reply:
Right. This affects Ads right now.